File: 2583 Kth Largest Sum in a Binary Tree/2583.py
class Solution:
    def kthLargestLevelSum(self, root: Optional[TreeNode], k: int) -> int:
        q, v = [root], []

        while q:
            next_q, total = [], 0

            for cur in q:
                if cur.left: next_q.append(cur.left)
                if cur.right: next_q.append(cur.right)
                total += cur.val
            
            v.append(total)
            q = next_q

        if k > len(v):
            return -1
        else:
            v.sort(reverse = True)
            return v[k-1]

File: 2583 Kth Largest Sum in a Binary Tree/2583.cpp
class Solution {
public:
    long long kthLargestLevelSum(TreeNode* root, int k) {
        priority_queue<long long> pq;
        queue<TreeNode*> q;

        q.push(root);

        while (!q.empty()){
            long long sum = 0;

            queue<TreeNode*> nextQ;

            while (!q.empty()){
                TreeNode* cur = q.front();
                q.pop();

                if (cur->left) nextQ.push(cur->left);
                if (cur->right) nextQ.push(cur->right);

                sum += cur->val;
            }

            pq.push(sum);
            q = nextQ;
        } 

        while (--k && !pq.empty()){
            pq.pop();
        }

        return pq.empty() ? -1 : pq.top();
    }
};
